BUG helicsMessageGetBytes truncates at NUL.

tomc-xsec opened the pull request:
helicsMessageGetBytes truncates the received data at the first
occurrence of NUL. This is because ffi.string(cdata, maxlen) scans maxlen bytes for the
first occurrence of NUL. Contrary to ffi.string, ffi.buffer(cdata, length) assumes cdata is length.

kdheepak commented:
Thanks for the PR, I didn’t realize that’s how `ffi.string` worked. 

Can you make a test for this?

tomc797 commented:
May I add the test to test_api:test_misc_api?

kdheepak commented:
Sure, that makes sense.

kdheepak commented:
I expect there may be other places where this change is necessary too. But I can go through the code and figure that out after this is merged.

tomc-xsec committed:
BUG helicsMessageGetBytes truncates at NUL.

helicsMessageGetBytes truncates the received data at the first
occurrence of NUL. ffi.string(cdata, maxlen) scans maxlen bytes for the
first occurrence of NUL. ffi.buffer(cdata, length) assumes length.

tomc-xsec committed:
Test binary data endpoint message handling.

Verify that endpoints can send and receive binary data messages,
messages which contain one or more NUL characters.

tomc797 commented:
I've added tests to cover the endpoints Bytes interfaces.

kdheepak commented:
Thanks a lot for the PR! That looks good to me. I’ll merge and make a new release this weekend.

kdheepak commented:
Thanks for the PR! I changed `ffi.buffer` to `ffi.unpack`:

https://github.com/GMLC-TDC/pyhelics/commit/e7c489995f49936fb16bde2e3f978f5758a699e1

kdheepak commented:
I made a release: https://github.com/GMLC-TDC/pyhelics/releases/tag/v3.0.1.post4

kdheepak commented:
Ah I didn’t realize that’s a new feature. Yeah I’ll bump it tonight.
